### PR TITLE
Allow more events to be buffered before the backend starts

### PR DESCRIPTION
**Pull request by ChenLingPeng**
Signed-off-by: forrestchen <forrestchen@tencent.com>

## Description
Our k8s cluster has more than 5000 nodes, chan with size 5000 will block flannel to run.

related issue: #719 

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by ChenLingPeng** [comment]
#729 adjust the size to 5000, and it is not enough now.

**Comment by rajatchopra** [comment]
Should we make this configurable? 
Through cli or env var, to the container in kube-flannel.yml daemonset.
Or, eventually when flannel has a helm chart based installation.

**Comment by ChenLingPeng** [comment]
@rajatchopra PR updated, please take a review

**Review by luthermonson — APPROVED** [review]

File: subnet/kube/kube.go [hunk 1]
```diff
@@ -138,7 +139,15 @@ func newKubeSubnetManager(c clientset.Interface, sc *subnet.Config, nodeName, pr
 	ksm.client = c
 	ksm.nodeName = nodeName
 	ksm.subnetConf = sc
-	ksm.events = make(chan subnet.Event, 5000)
+	scale := 5000
+	scaleStr := os.Getenv("K8S_SCALE")
+	if scaleStr != "" {
+		scale, _ := strconv.Atoi(scaleStr)
```
Comment (rajatchopra) [inline subnet/kube/kube.go:145]: We should be logging the error from Atoi?
Perhaps even return the error.

**Review by rajatchopra — CHANGES_REQUESTED** [review]

File: subnet/kube/kube.go (hunk 1 above)
Comment (ChenLingPeng) [reply subnet/kube/kube.go]: updated

File: subnet/kube/kube.go [hunk 2]
```diff
@@ -138,7 +139,18 @@ func newKubeSubnetManager(c clientset.Interface, sc *subnet.Config, nodeName, pr
 	ksm.client = c
 	ksm.nodeName = nodeName
 	ksm.subnetConf = sc
-	ksm.events = make(chan subnet.Event, 5000)
+	scale := 5000
+	scaleStr := os.Getenv("K8S_SCALE")
```
Comment (rajatchopra) [inline subnet/kube/kube.go:143]: How about EVENT_QUEUE_DEPTH instead  of K8S_SCALE?
We also want this env var documented. Here: https://github.com/flannel-io/flannel/blob/master/Documentation/configuration.md#environment-variables

**Comment by manuelbuil** [comment]
@ChenLingPeng will you have time to address the review?

**Comment by ChenLingPeng** [comment]
> @ChenLingPeng will you have time to address the review?

updated, PTAL

**Review by luthermonson — APPROVED** [review]

**Review by rbrtbnfgl — APPROVED** [review]

**Pull request merged**